samples/kafkaBasic/Consumer/ConsumerWorker.cs

```diff
@@ -0,0 +1,21 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Confluent.Kafka;
+
+namespace Consumer;
+internal sealed class ConsumerWorker(IConsumer<string, string> consumer, ILogger<ConsumerWorker> logger) : BackgroundService
+{
+    protected override Task ExecuteAsync(CancellationToken stoppingToken)
+    {
+        return Task.Factory.StartNew(() =>
+        {
+            consumer.Subscribe("topic");
+            while (!stoppingToken.IsCancellationRequested)
+            {
+                var result = consumer.Consume(stoppingToken);
+                logger.LogInformation($"Received message '{result.Message.Value}'.");
+            }
+        }, stoppingToken, TaskCreationOptions.LongRunning, TaskScheduler.Current);
```

Comment:
This API is very strange.. no async? Is it blocking?

Reply:
Indeed it is. See https://github.com/confluentinc/confluent-kafka-dotnet/blob/master/examples/Consumer/Program.cs and https://docs.confluent.io/kafka-clients/dotnet/current/overview.html#the-consume-loop

Reply:
Confluentic has bad [API](https://docs.confluent.io/platform/current/clients/confluent-kafka-dotnet/_site/api/Confluent.Kafka.IConsumer-2.html#Confluent_Kafka_IConsumer_2_Consume_CancellationToken_) here. It prefetches messages (if available), but it will block if nothing is there. That would be a perfect fit for `ValutTask` 😉.

Cf. https://github.com/confluentinc/confluent-kafka-dotnet/issues/487